cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -1978,12 +1985,15 @@ pub(crate) fn define(
     }
 
     // SIMD integer addition
-    let mut x86_padd_mapping: HashMap<LaneType, Vec<u8>> = HashMap::new();
-    x86_padd_mapping.insert(I8.into(), vec![0x66, 0x0f, 0xfc]); // PADDB from SSE2
-    x86_padd_mapping.insert(I16.into(), vec![0x66, 0x0f, 0xfd]); // PADDW from SSE2
-    x86_padd_mapping.insert(I32.into(), vec![0x66, 0x0f, 0xfe]); // PADDD from SSE2
-    x86_padd_mapping.insert(I64.into(), vec![0x66, 0x0f, 0xd4]); // PADDQ from SSE2
-    for (ty, opcodes) in x86_padd_mapping {
+    for (ty, opcodes) in [
+        (I8, vec![0x66, 0x0f, 0xfc]),  // PADDB from SSE2
```

Comment:
... but for this one, you can do:

```
    for &(ty, opcodes) in &[
        (I8, &[0x66, 0x0f, 0xfc]),  // PADDB from SSE2
        (I16, &[0x66, 0x0f, 0xfd]), // PADDW from SSE2
        (I32, &[0x66, 0x0f, 0xfe]), // PADDD from SSE2
        (I64, &[0x66, 0x0f, 0xd4]), // PADDQ from SSE2
    ]
    {
        let iadd = iadd.bind_vector_from_lane(ty, sse_vector_size);
        e.enc_32_64(iadd, rec_fa.opcodes(Vec::from(opcodes.as_ref())));
    }
```

(no more clon ing! To wit: it's possible to do so here because all the opcodes have the same length.)

Reply:
I still had to `ty.clone()` but I ended up using `opcodes.to_vec()` which is a bit shorter:
```
    for (ty, opcodes) in &[
        (I8, &[0x66, 0x0f, 0xfc]),  // PADDB from SSE2
        (I16, &[0x66, 0x0f, 0xfd]), // PADDW from SSE2
        (I32, &[0x66, 0x0f, 0xfe]), // PADDD from SSE2
        (I64, &[0x66, 0x0f, 0xd4]), // PADDQ from SSE2
    ] {
        let iadd = iadd.bind_vector_from_lane(ty.clone(), sse_vector_size);
        e.enc_32_64(iadd, rec_fa.opcodes(opcodes.to_vec()));
    }
```